app/javascript/components/CreateTransaction.jsx

```diff
@@ -0,0 +1,269 @@
+import PropTypes from "prop-types";
+import React from "react";
+
+/**
+* @prop artist: artist creating transaction
+* @prop buyer: buyer associated with transaction
+* @prop work: work associated with transaction
+*/
+
+class CreateTransaction extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      transaction: {
+        transaction_type: null,
+        start_date: null,
+        end_date: null,
+        purchase_date: null,
+        price: null,
+        buyer: null,
+        work: null,
+        comment: ""
+      },
+      types: null,
+      didMount: false,
+      formErrors: {
+        transaction_type: 'Please specify a type of transaction.',
+        purchase_date: 'Please choose a purchase date.',
+        start_date: 'Please choose a start date for the rental.',
+        end_date: 'Please choose an end date for the rental.',
+        price: 'Please enter a price for your transaction.'
+      },
+      fieldValid: {
+        type_valid: false,
+        purchase_valid: false,
+        start_valid: false,
+        end_valid: false,
+        price_valid: false,
+      },
+      formValid: false,
+      renderErrors: false,
+    }
+  }
+
+  /**
+  * TODO: error handling has to change - look at claire/toastr for example
+  */
+
+  handleChange = (event) => {
+    const transaction = this.state.transaction;
+    const name = event.target.name;
+    const value = event.target.value;
+    transaction[name] = value;
+    this.setState({ transaction: transaction },
+                  () => { this.validateField(name, value) });
+  }
+
+  validateField = (fieldName, value) => {
+    let errors = this.state.formErrors;
+    let fields = this.state.fieldValid;
+
+    switch(fieldName) {
+      case 'transaction_type':
+        fields.type_valid = value != null && value != 'choose';
+        errors.transaction_type = fields.type_valid ? '' : 'Please specify a type of transaction.'
+        break;
+      case 'purchase_date':
+        fields.purchase_valid = value != null && this.state.transaction.transaction_type == "purchase";
+        errors.purchase_date = fields.purchase_valid ? '' : errors.purchase_date
+        break;
+      case 'start_date':
+        fields.start_valid = value != null && this.state.transaction.transaction_type == "rental";
+        errors.start_date = fields.start_valid ? '' : errors.start_date
```

Comment:
this stuff is quite a pain to manage, looks fine to me tbh but we may need to refactor down the line
for now, this looks good to me